src/Microsoft.TestPlatform.Client/Execution/TestRunRequest.cs

```diff
@@ -84,6 +101,17 @@ public int ExecuteAsync()
             }
         }
 
+        internal void OnTestSessionTimeout(object obj)
+        {
+            string message = String.Format(ClientResources.TestSessionTimeoutMessage, this.testSessionTimeout);
+            var testMessagePayload = new CommunicationObjectModel.TestMessagePayload { MessageLevel = TestMessageLevel.Error, Message = message };
+            var rawMessage = this.dataSerializer.SerializePayload(CommunicationObjectModel.MessageType.TestMessage, testMessagePayload);
```

Comment:
This looks odd... We are serializing a message even though it is used in the same process again. Would be great to not do that. 
One more question: Why do we need both a `HandleLogMessage` and a `HandleRawMessage` ? Does not seem very evident since they both seem to be doing the same thing.